### PR TITLE
Persistent config path

### PR DESCRIPTION
**Pull request by audiobird**
Launching the program with a config file path creates a file ".prevconfpath" containing the path. On subsequent launches the program does not require a path argument. The previous path is used if none provided. 

Also added a startup script that prompts the user for a midi device id and virtual port number. The script accepts a config file path.

``` mctlstart.sh || mctlstart.sh  path/to/config```

Thanks a bunch for creating this program. It is immensely helpful!

### CODE REVIEW

File: src/midictl.c [hunk 1]
```diff
@@ -482,13 +482,50 @@ int main(int argc, char *argv[])
...
+	bool save_config_path = 1;
+	// check if config file path is given
+	//if not open the last given path.
+	if (config.config_path == NULL)
+	{
+		FILE *prev_config_path = fopen(".prevconfpath", "rt");
+		if (prev_config_path == NULL)
+		{
+			//prev file doesnt exist
+			fprintf(stderr, "Previously used config path is non existant\n");
```
Comment (Jacajack) [inline src/midictl.c]: I feel like this error message may be a little bit unclear when you run the program for the first time. Maybe something like `No config path specified and the previous config path file doesn't exists.` would be better? Printing the help message from `argp_usage()` here would be nice too, I think.

File: src/midictl.c [hunk 2]
```diff
@@ -482,13 +482,50 @@ int main(int argc, char *argv[])
...
+			exit(EXIT_FAILURE);
+		}
+		
+		//get the file size
+		fseek(prev_config_path, 0L, SEEK_END);
+    	uint32_t sz = ftell(prev_config_path) + 1;
+    	fseek(prev_config_path, 0L, SEEK_SET);
+
+		//get the path
+		char* f_path = malloc(sz);
```
Comment (Jacajack) [inline src/midictl.c:504]: Hardly significant, but it seems that you missed `free()` for this allocation. I guess the simplest solution here is to change `config.config_path` to `char*` and always store a dynamically allocated string there and always free it at the end.

Note that this code would need to use `strdup()` then:
https://github.com/Jacajack/midictl/blob/c0f11ac34ce451d4d6511eb426a1609cc6edb948/src/args.c#L37

File: src/midictl.c [hunk 3]
```diff
@@ -482,13 +482,50 @@ int main(int argc, char *argv[])
...
 	FILE *config_file = fopen(config.config_path, "rt");
 	if (config_file == NULL)
 	{
 		fprintf(stderr, "Could not open config file: %s\n", strerror(errno));
 		exit(EXIT_FAILURE);
 	}
+
+	if (save_config_path)
+	{
+		FILE *conf_path = fopen(".prevconfpath", "wt");
```
Comment (Jacajack) [inline src/midictl.c:525]: NULL check might be a good idea here. If `.prevconfpath` is a read-only file, `fopen()` will fail here and `fputs()` will be called with a null pointer.

**Review by Jacajack — CHANGES_REQUESTED** [review]
Hi and thanks for the pull request! Glad to hear you found midictl useful.

I like this idea, there are just a few minor changes I would make. It also seems that you accidentally deleted the `PKGBUILD` file. Otherwise than that, I think it's ready to merge.

Thanks again! :+1:

File: src/midictl.c (hunk 2 above)
Comment (audiobird) [reply src/midictl.c:504]: Admittedly this is my first time using malloc(). I went a different direction than what you said here. I'm also still new to pull requests. I've committed the change I made, but do I need to open a new pull request for you to be able to review? Sorry for my lack of github experience ;)

File: src/midictl.c (hunk 3 above)
Comment (audiobird) [reply src/midictl.c:525]: Good point there. Fixed!

File: src/midictl.c (hunk 1 above)
Comment (audiobird) [reply src/midictl.c]: I adjusted the error message to your suggestion. One thing though, I'm not familiar with the argp helper functions. I couldn't quickly figure out how I should call argp_usage() after the printf statement. I'll research and edit when I'm able to figure it out.

Comment (Jacajack) [reply src/midictl.c]: Oh, sorry, I forgot that `argp_usage()` takes a pointer to `argp_state`. That makes it a bit more tricky. You can try to find a workaround, but I think the new error message should be good enough.

File: src/midictl.c (hunk 2 above)
Comment (Jacajack) [reply src/midictl.c:504]: It's fine, I can see your changes here :)

**Comment by Jacajack** [comment]
Looks good to me! Please let me know if you would like to make any more changes, or if I can merge. I restored the `PKGBUILD` file in your repository, so you might want to run `git pull` before committing anything.

**Review by Jacajack — APPROVED** [review]

**Comment by audiobird** [comment]
I have no more changes to make, go ahead and merge! Thanks for the review and response!

**Comment by Jacajack** [comment]
Thanks!

**Pull request merged**